File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.cc
#include<iostream>

using namespace std;

int main() {

	int T;
	int sol[12] = { 0, };

	sol[1] = 1;
	sol[2] = 2;
	sol[3] = 4;

	cin >> T;

	for (int i = 4; i <= 10; i++) {
		sol[i] = sol[i - 1] + sol[i - 2] + sol[i - 3];
	}

	for (int i = 0; i < T; i++) {
		int n;

		cin >> n;

		cout << sol[n] << endl;
	}


	return 0;
}

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.java
import java.io.*;

public class Main {
    public static int sol=0;
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int T = Integer.parseInt(br.readLine());

        for (int i=0; i<T; i++) {
            sol = 0;
            int N = Integer.parseInt(br.readLine());
            rec(N, 0);

            bw.write(sol+"\n");
            bw.flush();
        }

        bw.close();

    }

    public static void rec(int n, int total) {
        if (total == n) {
            sol++;
            return;
        }

        for (int i=1; i<=3; i++) {
            if (total+i <= n) {
                rec(n, total+i);
            }
        }
    }
}